enterprise/frontend/src/embedding-sdk/store/auth.ts

```diff
@@ -0,0 +1,158 @@
+import type {
+  EmbeddingSessionToken,
+  FetchRequestTokenFn,
+  SDKConfig,
+} from "embedding-sdk";
+import { getIsLocalhost } from "embedding-sdk/lib/is-localhost";
+import type { SdkStoreState } from "embedding-sdk/store/types";
+import api from "metabase/lib/api";
+import { createAsyncThunk } from "metabase/lib/redux";
+import { refreshSiteSettings } from "metabase/redux/settings";
+import { refreshCurrentUser } from "metabase/redux/user";
+
+import { getOrRefreshSession } from "./reducer";
+import { getFetchRefreshTokenFn } from "./selectors";
+
+export const initAuth = createAsyncThunk(
+  "sdk/token/INIT_AUTH",
+  async (sdkConfig: SDKConfig, { dispatch }) => {
+    // Setup JWT or API key
+    const isValidJwtConfig =
+      sdkConfig.jwtProviderUri && sdkConfig.jwtProviderUri?.length > 0;
+    const isValidApiKeyConfig = sdkConfig.apiKey && getIsLocalhost();
+
+    if (isValidJwtConfig) {
+      // JWT setup
+      api.onBeforeRequest = async () => {
+        const session = await dispatch(
+          getOrRefreshSession(sdkConfig.jwtProviderUri!),
+        ).unwrap();
+        if (session?.id) {
+          api.sessionToken = session.id;
+        }
+      };
+      // verify that the session is actually valid before proceeding
+      await dispatch(getOrRefreshSession(sdkConfig.jwtProviderUri!)).unwrap();
+    } else if (isValidApiKeyConfig) {
+      // API key setup
+      api.apiKey = sdkConfig.apiKey;
+    }
+    // Fetch user and site settings
+    const [user, siteSettings] = await Promise.all([
+      dispatch(refreshCurrentUser()),
+      dispatch(refreshSiteSettings({})),
+    ]);
+
+    if (!user.payload) {
+      // The refresh user thunk just returns null if it fails to fetch the user, it doesn't throw
+      const error = new Error(
+        "Failed to fetch the user, is the session valid?",
```

Comment:
The error message feels a bit weird for a library. Should it sound something more official like this?

```suggestion
        "Failed to fetch the user, the session might be invalid.",
```

> [!note] The same thing could be said for the site setting error.